agent/engine/engine_integ_test.go

```diff
@@ -1202,3 +1202,50 @@ func TestStartStopWithCredentials(t *testing.T) {
 		t.Error("Credentials not removed from credentials manager for stopped task")
 	}
 }
+
+func TestStartStopWithSecurityOptionNoNewPrivileges(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	taskEvents, contEvents := taskEngine.TaskEvents()
+
+	testArn := "testSecurityOptionNoNewPrivileges"
+	testTask := createTestTask(testArn)
+	testTask.Containers[0].DockerConfig = api.DockerConfig{HostConfig: strptr(`{"SecurityOpt":["no-new-privileges"]}`)}
+
+	go taskEngine.AddTask(testTask)
+
+	found := false
+	for contEvent := range contEvents {
+		if contEvent.TaskArn != testTask.Arn {
+			continue
+		}
+		if contEvent.Status == api.ContainerRunning {
+			found = true
+			break
+		}
+		if contEvent.Status > api.ContainerRunning {
+			t.Error("Expect container to run not stop")
```

Comment:
This might loop forever if you leave this as `t.Error` instead of `t.Fatal`.  Can you change this to `t.Fatal`?  (Then you don't need the `found` variable either, since you'll only break out of the loop when it is successfully found.)